app/src/main/res/layout/fragment_main.xml

```diff
@@ -0,0 +1,103 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
```

Comment:
Android Studio has a nice thing called **Reformat Code** which works both in kotlin and xml files. Select all in this file, then `Code -> Reformat Code` from the menu (there is a keyboard shorcut too, probably it is _Ctrl+Alt+L_), you will see the difference. It is a good habit to reformat code before pushing.

The main problem in this layout is there is no space between Views, and its parameters are not ordered properly.